### PR TITLE
fix: code block content lost

### PR DESCRIPTION
**Pull request by frankcbliu**
As the issues say: https://github.com/NotionX/react-notion-x/issues/168

here i fix it like that:
image

### CODE REVIEW

File: packages/react-notion-x/src/block.tsx [hunk 1]
```diff
@@ -568,7 +572,7 @@ export const Block: React.FC<BlockProps> = (props) => {
 
     case 'code': {
       if (block.properties.title) {
-        const content = block.properties.title[0][0]
```
Comment (normdoow) [inline packages/react-notion-x/src/block.tsx:571]: Could you change it so there isn't a function call? I'd prefer it in one line.

**Comment by normdoow** [comment]
Interesting bug. My code block only have one title block with code in it. Not sure what makes it so there are multiple. Maybe Notion updated that as well.

**Comment by transitive-bullshit** [comment]
> Interesting bug. My code block only have one title block with code in it. Not sure what makes it so there are multiple. Maybe Notion updated that as well.

Every block has a version attached to it, so notion can internally make changes while remaining backwards compatible. I'd recommend that we try to always be defensive to try and support all block versions without having block version-specific logic (esp since it's really hard to test the diff btwn block versions).

File: packages/react-notion-x/src/block.tsx (hunk 1 above)
Comment (transitive-bullshit) [reply packages/react-notion-x/src/block.tsx:571]: Could be useful to have a `getBlockTitle` function in `notion-utils` that does this in a safe way. Not necessary, just a thought.

**Comment by normdoow** [comment]
> > Interesting bug. My code block only have one title block with code in it. Not sure what makes it so there are multiple. Maybe Notion updated that as well.
> 
> Every block has a version attached to it, so notion can internally make changes while remaining backwards compatible. I'd recommend that we try to always be defensive to try and support all block versions without having block version-specific logic (esp since it's really hard to test the diff btwn block versions).

This fix should do that pretty well then. As in work for all versions while not having to have version logic.

File: packages/react-notion-x/src/block.tsx (hunk 1 above)
Comment (frankcbliu) [reply packages/react-notion-x/src/block.tsx:571]: ok, i will fix it.

Comment (frankcbliu) [reply packages/react-notion-x/src/block.tsx:571]: > 

you means create a function such as named `getCodeContent` to do this?

Comment (normdoow) [reply packages/react-notion-x/src/block.tsx:571]: `getBlockTitle` would probably take a block as input and do this `block.properties.title.map((arr) => { return arr[0] }).join('')`
probably need to use ?'s or have some checks that the title is there. But I'm guessing  thats what he means.
I'm fine either way.

Comment (frankcbliu) [reply packages/react-notion-x/src/block.tsx:571]: > getBlockTitle would probably take a block as input and do this block.properties.title.map((arr) => { return arr[0] }).join('')
probably need to use ?'s or have some checks that the title is there. But I'm guessing thats what he means.
I'm fine either way.

i got it. but i found that the function `getBlockTitle` will lost some **rich text formatting**, the code:
image

of course, generally speaking, there are no such `rich text characters` in the code, so the problem should not be big

**Comment by frankcbliu** [comment]
maybe i broken the commit history, sorry for that, i will restart a new pull request.

**Pull request closed without merging**